R/rdeephaven/R/update_by_ops_wrapper.R

```diff
@@ -0,0 +1,615 @@
+#' @return UpdateByOp to be used in `update_by()`.
+#' @export
+udb_cum_min <- function(cols = character()) {
+  verify_string("cols", cols, FALSE)
+  return(UpdateByOp$new(INTERNAL_cum_min(cols)))
+}
+
+#' @description
+#' Creates a Cumulative Maximum UpdateByOp that computes the cumulative maximum of each column in `cols` for each aggregation group.
+#' @param cols String or list of strings denoting the column(s) to operate on. Can be renaming expressions, i.e. “new_col = col”.
+#' Default is to compute the cumulative maximum for all non-grouping columns.
+#' @return UpdateByOp to be used in `update_by()`.
+#' @export
+udb_cum_max <- function(cols = character()) {
+  verify_string("cols", cols, FALSE)
+  return(UpdateByOp$new(INTERNAL_cum_max(cols)))
+}
+
+#' @description
+#' Creates a Forward Fill UpdateByOp that replaces null values in `cols` with the last known non-null values.
+#' This operation is forward only.
+#' @param cols String or list of strings denoting the column(s) to operate on. Can be renaming expressions, i.e. “new_col = col”.
+#' Default is to perform a forward fill on all non-grouping columns.
+#' @return UpdateByOp to be used in `update_by()`.
+#' @export
+udb_forward_fill <- function(cols = character()) {
+  verify_string("cols", cols, FALSE)
+  return(UpdateByOp$new(INTERNAL_forward_fill(cols)))
+}
+
+#' @description
+#' Creates a Delta UpdateByOp for each column in `cols`. The Delta operation computes the difference between the
+#' current value and the previous value. When the current value is null, this operation will output null.
+#' When the current value is valid, the output will depend on the `delta_control` provided.
+#' @param cols String or list of strings denoting the column(s) to operate on. Can be renaming expressions, i.e. “new_col = col”.
+#' Default is to apply the delta operation to all non-grouping columns.
+#' @param delta_control Defines how the delta operation handles null values. Can be one of the following:
+#'   'null_dominates': A valid value following a null value returns null.
+#'   'value_dominates': A valid value following a null value returns the valid value.
+#'   'zero_dominates': A valid value following a null value returns zero.
+#' Defaults to `null_dominates`.
+#' @return UpdateByOp to be used in `update_by()`.
+#' @export
+udb_delta <- function(cols = character(), delta_control = "null_dominates") {
+  verify_string("cols", cols, FALSE)
+  if (!(delta_control %in% c("null_dominates", "value_dominates", "zero_dominates"))) {
+    stop(paste0("'delta_control' must be one of 'null_dominates', 'value_dominates', or 'zero_dominates'. Got '", delta_control, "'."))
+  }
+  return(UpdateByOp$new(INTERNAL_delta(cols, delta_control)))
+}
+
+#' @description
+#' Creates an exponential moving average UpdateByOp for each column in `cols`, using ticks as the decay unit.
+#' The formula used is
+#'   `a = e^(-1 / decay_ticks)`
+#'   `ema_next = a * ema_last + (1 - a) * value`
```

Comment:
I do not want to clutter these with the Latex syntax that the web docs are getting, because these aren't going to be seen in that format. I do think they could use the additional detail that the new docs provide.

Reply:
It does look like R supports latex formatting as well.  https://stackoverflow.com/questions/37397607/can-r-help-manuals-have-latex-math-in-them